text/0258-env-context-baggage-carriers.md

```diff
@@ -0,0 +1,236 @@
+# Environment Variable Specification for Context and Baggage Propagation
+
+This is a proposal to add Environment Variables to the OpenTelemetry
+specification as a carrier for context and baggage propagation between
+processes.
+
+## Table of Contents
+
+* [Motivation](#motivation)
+* [Explanation](#explanation)
+* [Internal Details](#internal-details)
+  * [UNIX](#unix)
+  * [Windows](#windows)
+  * [Allowed Characters](#allowed-characters)
+* [Trade-offs and Mitigations](#trade-offs-and-mitigations
+  * [Case-sensitivity](#case-sensitivity)
+  * [Security](#security)
+* [Prior Art and Alternatives](#prior-art-and-alternatives)
+  * [Alternatives and why they were not chosen](#alternatives-and-why-they-were-not-chosen)
+* [Open Questions](#open-questions)
+* [Future Possibilities](#future-possibilities)
+
+## Motivation
+
+The motivation for defining the specification for context and baggage
+propagation by using environment variables as carriers stems from a long open
+issue on the OpenTelemetry Specification repository, [issue #740](https://github.com/open-telemetry/opentelemetry-specification/issues/740#issue-665588273).
+
+The original issue identifies several use cases of systems that do not
+communicate across bounds by leveraging HTTP communications like, ETL, batch,
+and CI/CD systems, that need to benefit from context and baggage propagation.
+Defining a specification for this will enable better observability of these
+systems by enabling distributed tracing. There has already been a significant
+amount of [Prior Art](#prior-art) built within the industry and within
+OpenTelemetry to accomplish the immediate needs, however, OpenTelemetry at this
+time does not clearly define the specification for this form of propagation.
+
+Notably, as we define semantic conventions within the [CI/CD Working Group][cicd-wg],
+we'll need the specification defined for the industry to be able to adopt
+native tracing wtihin CI/CD systems.
+
+[cicd-wg]: https://github.com/open-telemetry/community/blob/main/projects/ci-cd.md
+
+## Explanation
+
+To propagate context and baggage between parent, and child processes
+in systems where HTTP communication does not occur, a specification for
+Environmental Variables can be used.
+
+Consider the following diagram:
+
+![Environment Variable Context Propagation](./img/0258-env-context-parent-child-process.png)
+
+In this diagram, a parent process is forked to spawn a child process,
```

Comment:
Why should parent be forked to spawn a child process? It's one possible implementation, but if I don't need to clone the parent memory space and just need to start a child process (e.g. via `os.system()` in Python), then it's not "forking".

Reply:
This diagram was simply showing an example and not language specific. It's simply a visual aid. If it needs to be adjusted with the different values to reflect possibilities, then I can work on that.

Reply:
Ok, I mistook the scope of the statements. I think forking is still an interesting use case, this comment https://github.com/open-telemetry/oteps/pull/258#discussion_r1616264598 still applies.

Reply:
Since the context propagation can use environment variables, a fork is not needed.

for example, you can run any command with the TRACEPARENT environment variable exported, then all processes in that shell will use the TRACEPARENT to propagate the context

```shell
export TRACEPARENT="00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01" 
my-new-process0
my-new-process1
my-new-process2
my-new-process3
```